pkg/vm.go

```diff
@@ -0,0 +1,113 @@
+package pkg
+
+import (
+	"fmt"
+	"net"
+	"os"
+)
+
+//go:generate zbusc -module vmd -version 0.0.1 -name manager -package stubs github.com/threefoldtech/zos/pkg+VMModule stubs/vmd_stub.go
+
+// VMNetworkInfo structure
+type VMNetworkInfo struct {
+	// Namespace name
+	Namespace string
+	// Tap device name
+	Tap string
+	// Mac address of the device
+	MAC net.HardwareAddr
+	// Address of the device in the form of cidr
+	AddressCIDR string
+	// Gateway gateway address
+	GatewayIP string
+	// Nameservers dns servers
+	Nameservers []string
+}
+
+// Disk specifies vm disk params
+type Disk struct {
+	// Size is disk size in Mib
+	Size uint64
+}
+
+// VM config structure
+type VM struct {
+	// virtual machine name, or ID
+	Name string
+	// CPU is number of cores assigned to the VM
+	CPU uint8
+	// Memory size in Mib
+	Memory int64
+	// Network is network info
+	Network VMNetworkInfo
+	// an allocated storage for the vm operation
+	// where files/virtual disks can be allocated
+	Storage string
```

Comment:
I'm not sure I understand what this fields represent. Is it a path where the vm disk file will be located ?

Reply:
Yeah, this will be gone after the changes to `storaged` but now it's basically the location where I do `truncate` of vm disks.